index.go

```diff
@@ -55,6 +64,10 @@ type Index interface {
 	SearchWithIDs(x []float32, k int64, include []int64, params json.RawMessage) (distances []float32,
 		labels []int64, err error)
 
+	// Applicable only to IVF indexes: Search clusters whose IDs are in eligibleCentroidIDs
+	SearchClusters(include, eligibleCentroidIDs []int64, minEligibleCentroids int,
```

Comment:
If this is applicable to IVFIndexes, let's update the name of this one as well - `SearchClustersFromIVFIndex`